src/lib/leaderboard/trpc.ts

```diff
@@ -0,0 +1,27 @@
+
+import { createTRPCRouter, publicProcedure } from "~/lib/trpc/trpc"
+import { orderBy, schema } from "../db"
+import { z } from "zod"
+
+export const leaderboardRouter = createTRPCRouter({
+  getLeaderboard: publicProcedure.input(z.object({
+    limit: z.number().min(1).max(10),
+    cursor: z.number().nullish(),
+  })).query(async ({ ctx, input }) => {
+    const limit = input.limit ?? 10;
+    const { cursor } = input;
+
+    const users = await ctx.db.query.users.findMany({
+      orderBy: [orderBy.desc(schema.users.wins), orderBy.asc(schema.users.name)],
+      limit: limit + 1,
+      offset: cursor ?? 0,
+    })
+
+    const hasNextPage = users.length > limit;
+
+    return {
+      users: hasNextPage ? users.slice(0, limit) : users,
```

Comment:
shouldn't we be always returning `users`?

Reply:
Im not doing infinite scroll anymore, but for completeness I fetch 1 more record than I need just to check if there are anymore users to fetch, so if there are I only want to return the `limit` number of users not `limit + 1`, and if there are less than the limit number of users just return the users. 
Not sure if this is the best way to be doing it but thats why I have this conditional here